pandas/io/parsers/c_parser_wrapper.py

```diff
@@ -265,7 +293,76 @@ def _get_index_names(self):
 
         return names, idx_names
 
-    def _maybe_parse_dates(self, values, index, try_parse_dates=True):
+    def _maybe_parse_dates(self, values, index: int, try_parse_dates=True):
         if try_parse_dates and self._should_parse_dates(index):
             values = self._date_conv(values)
         return values
+
+
+def _concatenate_chunks(chunks: list[dict[int, ArrayLike]]) -> dict:
+    """
+    Concatenate chunks of data read with low_memory=True.
+
+    The tricky part is handling Categoricals, where different chunks
+    may have different inferred categories.
+    """
+    names = list(chunks[0].keys())
+    warning_columns = []
+
+    result = {}
+    for name in names:
```

Comment:
do we not have a generic pandas concat call to handle this? or is that for a followup

Reply:
> do we not have a generic pandas concat call to handle this? or is that for a followup

im hopeful we can use something more generic, but havent tried it out yet (comment on L336)